### PR TITLE
Fixes return value of i2s_stm32_write by returning queue_put() return and fix I2S format channel count.

### PR DESCRIPTION
**Pull request by cdesjardins**
Fixes return value of i2s_stm32_write by returning queue_put() return and fix I2S format channel count.

### CODE REVIEW

**Review by marwaiehm-st — DISMISSED** [review]
LGTM

File: drivers/i2s/i2s_ll_stm32.c [hunk 1]
```diff
@@ -182,7 +182,8 @@ static int i2s_stm32_configure(const struct device *dev, enum i2s_dir dir,
 	 * When I2S data format is selected parameter channels is ignored,
 	 * number of words in a frame is always 2.
 	 */
-	const uint32_t num_channels = i2s_cfg->format & I2S_FMT_DATA_FORMAT_MASK
+	const uint32_t num_channels = ((i2s_cfg->format & I2S_FMT_DATA_FORMAT_MASK)	==
+				      I2S_FMT_DATA_FORMAT_I2S)
```
Comment (erwango) [inline drivers/i2s/i2s_ll_stm32.c:186]:
```suggestion
	const uint32_t num_channels = ((i2s_cfg->format & I2S_FMT_DATA_FORMAT_MASK) ==
				      I2S_FMT_DATA_FORMAT_I2S) ? 2U : i2s_cfg->channels;
```

**Review by erwango — CHANGES_REQUESTED** [review]
@cdesjardins Thanks for this contribution.
Content of the change looks ok, but one comment on coding style and also please fix compliance issues reported by bot

File: drivers/i2s/i2s_ll_stm32.c (hunk 1 above)
Comment (bas-archembedded) [reply drivers/i2s/i2s_ll_stm32.c]: @erwango I have addressed most of these issues for @cdesjardins, my colleage ;)

**Review by erwango — APPROVED** [review]

**Review by marwaiehm-st — APPROVED** [review]

**Pull request merged**